packages/plugin-cloudflare/src/tunnel.ts

```diff
@@ -0,0 +1,87 @@
+import {TUNNEL_PROVIDER} from './provider.js'
+import {startTunnel, TunnelError} from '@shopify/cli-kit/node/plugins/tunnel'
+import {err, ok, Result} from '@shopify/cli-kit/node/result'
+import {exec, captureOutput} from '@shopify/cli-kit/node/system'
+import {joinPath, dirname} from '@shopify/cli-kit/node/path'
+import {Writable} from 'stream'
+import {fileURLToPath} from 'url'
+
+export default startTunnel({provider: TUNNEL_PROVIDER, action: hookStart})
+
+export type ReturnType = Promise<Result<{url: string}, TunnelError>>
+
+export async function hookStart(port: number): ReturnType {
+  try {
+    const {url} = await tunnel({port})
+    return ok({url})
+    // eslint-disable-next-line no-catch-all/no-catch-all, @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    const tunnelError = new TunnelError('unknown', "Couldn't start tunnel")
```

Comment:
Some suggestions here:
- I think we can remove the `Couldn't start tunnel`, as we are also showing `cloudflare failed to start the tunnel.`
- I'd add a "try this" section with some suggestions like:
  - Try again
  - Use ngrok with this flag
  - Pass a custom tunnel URL this way
- The stacktrace is probably not very useful here
- For some reason, the original output from cloudflare is appearing below the error banner when it crashes

![27-08-671i7-spzqq](https://user-images.githubusercontent.com/14979109/221631637-e48d0c08-6267-4ad5-9f2e-cce3e591e025.png)

Reply:
About the output, that's how we treat "external" errors, when an external tool crashes we showing the message reported from that tool...
you can see that on line 75 of `public/node/system.ts`

Should we treat this as a special case? not sure how we can do that 🤔

Reply:
I see. Ideally, I think we could include it in the error banner above, maybe instead of the stacktrace.

Reply:
I removed the stacktrace from that error, but I don't think we can control how to include the other part 🤔 
Let's try and use it for some time and see how it feels and we can discuss how to improve the DX